### ISSUE TITLE
Parameter for number of cells in SimpleCcBattery module

### ISSUE THREAD

**Issue by n-jay** [inet-framework/inet#843]
I've been working with a simulation that involves the `SimpleCcBattery` [module](https://github.com/inet-framework/inet/blob/master/src/inet/power/storage/SimpleCcBattery.ned) and noticed the lack of a **number of cells** parameter.
Considering we have to manually calculate and individually amend each of the other parameters (Capacity, Voltage and Resistance) whenever more than one cell is used (eg: 2 AA batteries in sensor node) it would be ideal if a **number of cells** parameter was also present.

I'm happy to develop and contribute this feature and am creating this issue to get feedback before committing my time to this.

**Comment by levy** [comment]
Can you be please more specific what parameterization are you suggeting? I have a feeling that it would be best to implement this in a derived module called AaBatteryPack or something.

**Comment by n-jay** [comment]
> it would be ideal if a number of cells parameter was also present.

I meant a NED parameter (with necessary amendments on C++ side to accommodate). Its default value would be 1. 
Wouldn't a completely new derived module be a bit of overkill just for one extra parameter?

**Comment by levy** [comment]
You don't need C++ code to do that, so a new NED module would be ok to me. It perfectly captures the idea of combining AA batteries in pack. Although there are many ways to do that, e.g. parallel vs. serial connections, etc. Having such parameters in the SimpleCcBattery looks wrong to me.

**Comment by n-jay** [comment]
> ...so a new NED module would be ok to me. It perfectly captures the idea of combining AA batteries in pack.

Ack. Will create it as a new module in that case.
Since I'm not very familiar with derived modules, could you link an existing derived module that I can perhaps refer to get an idea?

> You don't need C++ code to do that, so a new NED module would be ok to me.
...Although there are many ways to do that, e.g. parallel vs. serial connections, etc.

I was under the impression that we'll need C++ coding to calculate the final current & voltage values (based on the values ingested by the NED parameters), especially depending on whether the cells are arranged serially or parallelly.

**Comment by levy** [comment]
There are plenty of examples in INET. Also, the implementation can be done in NED to a certain extent depending on what you really want to achieve.

**Comment by n-jay** [comment]
I see. I'll create the first version and submit as a PR. I suppose the necessary amendments can be made thereafter based on feedback.
Shall I keep this issue open so the PR can be linked once ready?

**Comment by levy** [comment]
Yep.

**Comment by n-jay** [comment]
Hi @levy apologies for the delay. 
I've started development on this. Is it ok if I continue development since you've self-assigned this?

**Comment by levy** [comment]
It's ok, I don't plan to do anything with this issue, other than reviewing a PR if there is one.

**Comment by n-jay** [comment]
> It's ok, I don't plan to do anything with this issue, other than reviewing a PR if there is one.

Noted @levy 
I created the first version of this as a PR: https://github.com/inet-framework/inet/pull/850.

For the moment I implemented a series pack since that's the most straightforward. 
A parallel implementation would involve the `energyConsumer` module as well since current become additive and also I wasn't quite sure how to write parameter variable conditions in the NED file.

Hope to make amendments to the PR based on feedback.
